File: replace-tailwind.py
import os
import re

def replace_classes(content):
    # Margins
    content = re.sub(r'\bml-([0-9a-z.-]+)', r'ms-\1', content)
    content = re.sub(r'\bmr-([0-9a-z.-]+)', r'me-\1', content)
    content = re.sub(r'-ml-([0-9a-z.-]+)', r'-ms-\1', content)
    content = re.sub(r'-mr-([0-9a-z.-]+)', r'-me-\1', content)
    # Paddings
    content = re.sub(r'\bpl-([0-9a-z.-]+)', r'ps-\1', content)
    content = re.sub(r'\bpr-([0-9a-z.-]+)', r'pe-\1', content)
    # Text alignment
    content = re.sub(r'\btext-left\b', 'text-start', content)
    content = re.sub(r'\btext-right\b', 'text-end', content)
    # Borders
    content = re.sub(r'\bborder-l-([0-9a-z.-]+)', r'border-s-\1', content)
    content = re.sub(r'\bborder-r-([0-9a-z.-]+)', r'border-e-\1', content)
    content = re.sub(r'\bborder-l\b', 'border-s', content)
    content = re.sub(r'\bborder-r\b', 'border-e', content)
    # BorderRadius
    content = re.sub(r'\brounded-l-([0-9a-z.-]+)', r'rounded-s-\1', content)
    content = re.sub(r'\brounded-r-([0-9a-z.-]+)', r'rounded-e-\1', content)
    content = re.sub(r'\brounded-l\b', 'rounded-s', content)
    content = re.sub(r'\brounded-r\b', 'rounded-e', content)
    # Positioning
    content = re.sub(r'\bleft-([0-9a-z.-]+)', r'start-\1', content)
    content = re.sub(r'\bright-([0-9a-z.-]+)', r'end-\1', content)
    content = re.sub(r'-left-([0-9a-z.-]+)', r'-start-\1', content)
    content = re.sub(r'-right-([0-9a-z.-]+)', r'-end-\1', content)
    content = re.sub(r'\bleft-0\b', 'start-0', content)
    content = re.sub(r'\bright-0\b', 'end-0', content)
    return content

for directory in ['resources/views', 'resources/js']:
    for root, _, files in os.walk(directory):
        for file in files:
            if file.endswith('.blade.php') or file.endswith('.js') or file.endswith('.vue'):
                filepath = os.path.join(root, file)
                with open(filepath, 'r+', encoding='utf-8') as f:
                    content = f.read()
                    new_content = replace_classes(content)
                    if content != new_content:
                        f.seek(0)
                        f.write(new_content)
                        f.truncate()
                        print(f"Updated {filepath}")

File: replace-tailwind.mjs
import fs from 'fs';
import path from 'path';

function walkDir(dir, callback) {
    fs.readdirSync(dir).forEach(f => {
        let dirPath = path.join(dir, f);
        let isDirectory = fs.statSync(dirPath).isDirectory();
        isDirectory ? walkDir(dirPath, callback) : callback(path.join(dir, f));
    });
}

function replaceClasses(content) {
    return content
        // Margins
        .replace(/\bml-([0-9a-z\.-]+)/g, 'ms-$1')
        .replace(/\bmr-([0-9a-z\.-]+)/g, 'me-$1')
        .replace(/-ml-([0-9a-z\.-]+)/g, '-ms-$1')
        .replace(/-mr-([0-9a-z\.-]+)/g, '-me-$1')
        // Paddings
        .replace(/\bpl-([0-9a-z\.-]+)/g, 'ps-$1')
        .replace(/\bpr-([0-9a-z\.-]+)/g, 'pe-$1')
        // Text alignment
        .replace(/\btext-left\b/g, 'text-start')
        .replace(/\btext-right\b/g, 'text-end')
        // Borders
        .replace(/\bborder-l-([0-9a-z\.-]+)/g, 'border-s-$1')
        .replace(/\bborder-r-([0-9a-z\.-]+)/g, 'border-e-$1')
        .replace(/\bborder-l\b/g, 'border-s')
        .replace(/\bborder-r\b/g, 'border-e')
        // BorderRadius
        .replace(/\brounded-l-([0-9a-z\.-]+)/g, 'rounded-s-$1')
        .replace(/\brounded-r-([0-9a-z\.-]+)/g, 'rounded-e-$1')
        .replace(/\brounded-l\b/g, 'rounded-s')
        .replace(/\brounded-r\b/g, 'rounded-e')
        // Positioning
        .replace(/\bleft-([0-9a-z\.-]+)/g, 'start-$1')
        .replace(/\bright-([0-9a-z\.-]+)/g, 'end-$1')
        .replace(/-left-([0-9a-z\.-]+)/g, '-start-$1')
        .replace(/-right-([0-9a-z\.-]+)/g, '-end-$1')
        .replace(/\bleft-0\b/g, 'start-0')
        .replace(/\bright-0\b/g, 'end-0');
}

['resources/views', 'resources/js'].forEach(dir => {
    if (!fs.existsSync(dir)) return;
    walkDir(dir, (filePath) => {
        if (filePath.endsWith('.blade.php') || filePath.endsWith('.js') || filePath.endsWith('.vue')) {
            let content = fs.readFileSync(filePath, 'utf8');
            let newContent = replaceClasses(content);
            if (content !== newContent) {
                fs.writeFileSync(filePath, newContent);
                console.log(`Updated ${filePath}`);
            }
        }
    });
});
